eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Main.java

```diff
@@ -30,35 +32,48 @@ private static BomGenerator parseCommandLine(String[] args) {
 
                     switch (argName.toLowerCase()) {
                         case COMMANDLINE_INPUTFILE:
-                            inputFile = argValue;
+                            validateNotNullOrEmpty(argName, argValue);
+                            generator.setInputFileName(argValue);
                             break;
 
                         case COMMANDLINE_OUTPUTFILE:
-                            outputFile = argValue;
+                            validateNotNullOrEmpty(argName, argValue);
+                            generator.setOutputFileName(argValue);
                             break;
 
                         case COMMANDLINE_POMFILE:
-                            pomFile = argValue;
+                            validateNotNullOrEmpty(argName, argValue);
+                            generator.setPomFileName(argValue);
+                            break;
+
+                        case COMMANDLINE_MODE:
+                            validateNotNullOrEmpty(argName, argValue);
+                            validateValues(argName, argValue, GENERATE_MODE, ANALYZE_MODE);
+                            generator.setMode(argValue);
                             break;
                     }
                 }
             }
         }
 
-        // validate that each of these are present.
-        validateInputs(inputFile, outputFile, pomFile);
-        return new BomGenerator(inputFile, outputFile, pomFile);
+        validateOptions(generator);
     }
 
-    private static void validateInputs(String inputFile, String outputFile, String pomFile) {
-        validateInput(inputFile, COMMANDLINE_INPUTFILE);
-        validateInput(outputFile, COMMANDLINE_OUTPUTFILE);
-        validateInput(pomFile, COMMANDLINE_POMFILE);
-    }
+    private static void validateOptions(BomGenerator generator) {
+        switch (generator.getMode()) {
+            case ANALYZE_MODE:
+                // In analyze mode, we should only be setting the pom file.
+                validateNotNullOrEmpty(generator.getPomFileName(), "pomFile");
+                validateNull(generator.getInputFileName(), "inputFileName");
+                validateNull(generator.getOutputFileName(), "outputFileName");
```

Comment:
Could these just be ignored instead of erroring?

Reply:
Sure, changing it :).